virtualenv related error failing Debian package builds

```
PKG_NAME="securedrop-log" ./scripts/build-debianpackage
gpg: key 310F561200F4AD77: public key "SecureDrop Release Signing Key" imported
gpg: Total number processed: 1
gpg:               imported: 1
gpg: Signature made Tue Feb  4 22:51:10 2020 UTC
gpg:                using RSA key 22245C81E3BAEB4138B36061310F561200F4AD77
gpg: Good signature from "SecureDrop Release Signing Key" [unknown]
gpg:                 aka "SecureDrop Release Signing Key <securedrop-release-key@freedom.press>" [unknown]
gpg: WARNING: This key is not certified with a trusted signature!
gpg:          There is no indication that the signature belongs to the owner.
Primary key fingerprint: 2224 5C81 E3BA EB41 38B3  6061 310F 5612 00F4 AD77
Checking that SHA256SUMs from mirror match signed file...  OK
securedrop-log is a Python package
securedrop-log-1000.0/LICENSE
securedrop-log-1000.0/MANIFEST.in
securedrop-log-1000.0/PKG-INFO
securedrop-log-1000.0/README.md
securedrop-log-1000.0/VERSION
securedrop-log-1000.0/build-requirements.txt
securedrop-log-1000.0/changelog.md
securedrop-log-1000.0/requirements.txt
securedrop-log-1000.0/sd-rsyslog
securedrop-log-1000.0/sd-rsyslog-example.conf
securedrop-log-1000.0/sdlog.conf
securedrop-log-1000.0/securedrop-log
securedrop-log-1000.0/securedrop-log-saver
securedrop-log-1000.0/securedrop-log.service
securedrop-log-1000.0/securedrop-redis-log
securedrop-log-1000.0/securedrop.Log
securedrop-log-1000.0/securedrop_log.egg-info/
securedrop-log-1000.0/securedrop_log.egg-info/PKG-INFO
securedrop-log-1000.0/securedrop_log.egg-info/SOURCES.txt
securedrop-log-1000.0/securedrop_log.egg-info/dependency_links.txt
securedrop-log-1000.0/securedrop_log.egg-info/top_level.txt
securedrop-log-1000.0/setup.cfg
securedrop-log-1000.0/setup.py
All hashes verified.
Building package 'securedrop-log' from version '1000.0'...
/home/circleci/debbuild/packaging/securedrop-log/
dpkg-buildpackage: info: source package securedrop-log
dpkg-buildpackage: info: source version 0.0.4+buster
dpkg-buildpackage: info: source distribution unstable
dpkg-buildpackage: info: source changed by Kushal Das <kushal@freedom.press>
dpkg-buildpackage: info: host architecture amd64
 dpkg-source --before-build .
 fakeroot debian/rules clean
make[1]: Entering directory '/home/circleci/debbuild/packaging/securedrop-log'
dh clean --with python-virtualenv --python /usr/bin/python3 --setuptools --extra-pip-arg "--no-deps" --index-url https://pypi.securedrop.org/simple --requirements build-requirements.txt
   dh_clean -O--python=/usr/bin/python3 -O--setuptools -O--extra-pip-arg -O--no-deps -O--index-url=https://pypi.securedrop.org/simple -O--requirements=build-requirements.txt
make[1]: Leaving directory '/home/circleci/debbuild/packaging/securedrop-log'
 dpkg-source -b .
dpkg-source: warning: no source format specified in debian/source/format, see dpkg-source(1)
dpkg-source: info: using source format '1.0'
dpkg-source: warning: source directory 'securedrop-log' is not <sourcepackage>-<upstreamversion> 'securedrop-log-0.0.4+buster'
dpkg-source: info: building securedrop-log in securedrop-log_0.0.4+buster.tar.gz
dpkg-source: info: building securedrop-log in securedrop-log_0.0.4+buster.dsc
 debian/rules build
make[1]: Entering directory '/home/circleci/debbuild/packaging/securedrop-log'
dh build --with python-virtualenv --python /usr/bin/python3 --setuptools --extra-pip-arg "--no-deps" --index-url https://pypi.securedrop.org/simple --requirements build-requirements.txt
   dh_update_autotools_config -O--python=/usr/bin/python3 -O--setuptools -O--extra-pip-arg -O--no-deps -O--index-url=https://pypi.securedrop.org/simple -O--requirements=build-requirements.txt
   dh_auto_configure -O--python=/usr/bin/python3 -O--setuptools -O--extra-pip-arg -O--no-deps -O--index-url=https://pypi.securedrop.org/simple -O--requirements=build-requirements.txt
dh_auto_configure: Please use the third-party "pybuild" build system instead of python-distutils
dh_auto_configure: This feature will be removed in compat 12.
make[1]: Leaving directory '/home/circleci/debbuild/packaging/securedrop-log'
 fakeroot debian/rules binary
make[1]: Entering directory '/home/circleci/debbuild/packaging/securedrop-log'
dh binary --with python-virtualenv --python /usr/bin/python3 --setuptools --extra-pip-arg "--no-deps" --index-url https://pypi.securedrop.org/simple --requirements build-requirements.txt
   dh_testroot -O--python=/usr/bin/python3 -O--setuptools -O--extra-pip-arg -O--no-deps -O--index-url=https://pypi.securedrop.org/simple -O--requirements=build-requirements.txt
   dh_prep -O--python=/usr/bin/python3 -O--setuptools -O--extra-pip-arg -O--no-deps -O--index-url=https://pypi.securedrop.org/simple -O--requirements=build-requirements.txt
   dh_install -O--python=/usr/bin/python3 -O--setuptools -O--extra-pip-arg -O--no-deps -O--index-url=https://pypi.securedrop.org/simple -O--requirements=build-requirements.txt
   dh_installdocs -O--python=/usr/bin/python3 -O--setuptools -O--extra-pip-arg -O--no-deps -O--index-url=https://pypi.securedrop.org/simple -O--requirements=build-requirements.txt
   dh_installchangelogs -O--python=/usr/bin/python3 -O--setuptools -O--extra-pip-arg -O--no-deps -O--index-url=https://pypi.securedrop.org/simple -O--requirements=build-requirements.txt
   dh_virtualenv -O--python=/usr/bin/python3 -O--setuptools -O--extra-pip-arg -O--no-deps -O--index-url=https://pypi.securedrop.org/simple -O--requirements=build-requirements.txt
usage: virtualenv [--version] [-v | -q] [--discovery {builtin}] [-p py] [--creator {builtin,cpython3-posix,venv}] [--seeder {app-data,pip}] [--no-seed] [--activators comma_separated_list] [--clear] [--system-site-packages]
                  [--symlinks | --copies] [--download | --no-download] [--extra-search-dir d [d ...]] [--pip version] [--setuptools version] [--wheel version] [--no-pip] [--no-setuptools] [--no-wheel] [--clear-app-data] [--prompt prompt]
                  [-h]
                  [dest]
virtualenv: error: argument --setuptools: expected one argument
Traceback (most recent call last):
  File "/usr/bin/dh_virtualenv", line 110, in <module>
    sys.exit(main() or 0)
  File "/usr/bin/dh_virtualenv", line 87, in main
    deploy.create_virtualenv()
  File "/usr/lib/python2.7/dist-packages/dh_virtualenv/deployment.py", line 157, in create_virtualenv
    subprocess.check_call(virtualenv)
  File "/usr/lib/python2.7/subprocess.py", line 190, in check_call
    raise CalledProcessError(retcode, cmd)
subprocess.CalledProcessError: Command '['virtualenv', '--no-site-packages', '--setuptools', '--python', '/usr/bin/python3', 'debian/securedrop-log/opt/venvs/securedrop-log']' returned non-zero exit status 2
make[1]: *** [debian/rules:4: binary] Error 1
make[1]: Leaving directory '/home/circleci/debbuild/packaging/securedrop-log'
dpkg-buildpackage: error: fakeroot debian/rules binary subprocess returned exit status 2
make: *** [Makefile:32: securedrop-log] Error 2
```

--- a/.circleci/config.yml
+++ b/.circleci/config.yml
@@ -1,5 +1,10 @@
 ---
 common-steps:
+  - &removevirtualenv
+    run:
+      name: Removes the upstream virtualenv from the original container image
+      command: sudo pip uninstall virtualenv -y
+
   - &installdeps
     run:
       name: Install Debian packaging dependencies
@@ -161,15 +166,15 @@ common-steps:
         echo 'export PKG_NAME=$(cat ~/packaging/sd_package_name)' >> $BASH_ENV
 
   - &setmetapackageversion
-      run:
-        name: Get metapackage version via distribution changelog
-        command: |
-          PLATFORM="$(lsb_release -sc)"
-          CURRENT_VERSION=$(grep -oP "\d+\.\d+\.\d+" ${PKG_NAME}/debian/changelog-${PLATFORM} | head -n1)
-          export VERSION_TO_BUILD="$CURRENT_VERSION"
-          # Enable access to this env var in subsequent run steps
-          echo $VERSION_TO_BUILD > ~/packaging/sd_version
-          echo 'export VERSION_TO_BUILD=$(cat ~/packaging/sd_version)' >> $BASH_ENV
+    run:
+      name: Get metapackage version via distribution changelog
+      command: |
+        PLATFORM="$(lsb_release -sc)"
+        CURRENT_VERSION=$(grep -oP "\d+\.\d+\.\d+" ${PKG_NAME}/debian/changelog-${PLATFORM} | head -n1)
+        export VERSION_TO_BUILD="$CURRENT_VERSION"
+        # Enable access to this env var in subsequent run steps
+        echo $VERSION_TO_BUILD > ~/packaging/sd_version
+        echo 'export VERSION_TO_BUILD=$(cat ~/packaging/sd_version)' >> $BASH_ENV
 
   - &installgitlfs
     run:
@@ -250,6 +255,7 @@ jobs:
       - image: circleci/python:3.7-buster
     steps:
       - checkout
+      - *removevirtualenv
       - *installdeps
       - *clonesecuredroplog
       - *getlatestreleasedversion
@@ -261,6 +267,7 @@ jobs:
       - image: circleci/python:3.7-buster
     steps:
       - checkout
+      - *removevirtualenv
       - *installdeps
       - *clonesecuredroplog
       - *getnightlyversion
@@ -275,6 +282,7 @@ jobs:
       - image: circleci/python:3.7-buster
     steps:
       - checkout
+      - *removevirtualenv
       - *installdeps
       - *clonesecuredropclient
       - *getlatestreleasedversion
@@ -286,6 +294,7 @@ jobs:
       - image: circleci/python:3.7-buster
     steps:
       - checkout
+      - *removevirtualenv
       - *installdeps
       - *clonesecuredropclient
       - *getnightlyversion
@@ -300,6 +309,7 @@ jobs:
       - image: circleci/python:3.7-buster
     steps:
       - checkout
+      - *removevirtualenv
       - *installdeps
       - *clonesecuredropproxy
       - *getlatestreleasedversion
@@ -311,6 +321,7 @@ jobs:
       - image: circleci/python:3.7-buster
     steps:
       - checkout
+      - *removevirtualenv
       - *installdeps
       - *clonesecuredropproxy
       - *getnightlyversion
@@ -325,6 +336,7 @@ jobs:
       - image: circleci/python:3.7-buster
     steps:
       - checkout
+      - *removevirtualenv
       - *installdeps
       - *clonesecuredropexport
       - *getlatestreleasedversion
@@ -336,6 +348,7 @@ jobs:
       - image: circleci/python:3.7-buster
     steps:
       - checkout
+      - *removevirtualenv
       - *installdeps
       - *clonesecuredropexport
       - *getnightlyversion
@@ -350,6 +363,7 @@ jobs:
       - image: circleci/python:3.7-buster
     steps:
       - checkout
+      - *removevirtualenv
       - *installdeps
       - *setsvsdispname
       - *setmetapackageversion
@@ -360,6 +374,7 @@ jobs:
       - image: circleci/python:3.7-buster
     steps:
       - checkout
+      - *removevirtualenv
       - *installdeps
       - *setsvsdispname
       - *getnightlymetapackageversion
@@ -414,9 +429,9 @@ workflows:
       - build-buster-securedrop-workstation-grsec
       - make-dom0-rpm
 
-# Nightly jobs for each package are run in series to ensure there are no
-# conflicts or race conditions when committing deb packages to git-lfs.
-# Each nightly job requires the completion of the previous task in the sequence.
+  # Nightly jobs for each package are run in series to ensure there are no
+  # conflicts or race conditions when committing deb packages to git-lfs.
+  # Each nightly job requires the completion of the previous task in the sequence.
   nightly:
     triggers:
       - schedule:

